crypto/evp/e_aes.c

```diff
@@ -115,6 +115,14 @@ void vpaes_cbc_encrypt(const unsigned char *in,
                        unsigned char *out,
                        size_t length,
                        const AES_KEY *key, unsigned char *ivec, int enc);
+# if defined(__x86_64) || defined(__x86_64__) \
+     || defined(_M_AMD64) || defined(_M_X64)
```

Comment:
Not for this PR, but grepping for `__x86_64` in the source tree, I see a lot of variations of this ifdef. Perhaps there's something to be said for having some common architecture ifdefs in some common file?

Reply:
I don't even know which compiler made this mess necessary.
I could find out for GCC, but there are lots of other compilers
where I cannot look at the source code.  I do not understand
if that was just s stupid idea from the beginning. or was there
a reason for this multiple defines?  There is no comment
anywhere, giving the rationale, therefore I personally think
that was driven by irrational fear about possible bugs in
unknown / ancient no longer supported compiler versions.